{"name": "test_api_audit[case1]", "status": "failed", "statusDetails": {"message": "AssertionError: assert '项目不在审核中状态' == '项目不存在'\n  - 项目不存在\n  + 项目不在审核中状态", "trace": "self = <tests.test_audit.Test_Audit object at 0x0000016EF719A128>\ncase = {'Headers': '{\"Content-Type\":\"application/json\",\"X-Lemonban-Media-Type\":\"lemonban.v2\",\"Authorization\":None}', 'auto_script': '项目不存在', 'case_id': 2, 'data': '{\\n    \"loan_id\":\"#N_id#\",\\n    \"approved_or_not\":\"true\"\\n\\n}', ...}\nget_root_token = ({'N_id': 2064567, 'get_id': 2037397, 'give_id': 2052534, 'no_id': 2037790, ...}, {'shenhe': 'Bearer eyJhbGciOiJIUzUxM...5MH0.pIjGHVqRQZEKWvAUBcTKfCPSn2MA2ONc_tmAliMBDgS24mJBnQJIt1F8kN_Qy6vIS2sw_xm1sALG9cyEMzXp-Q', 'shenhe_id': 1000337288})\nget_token = ('Bearer eyJhbGciOiJIUzUxMiJ9.eyJtZW1iZXJfaWQiOjEwMDAzMzcyNjUsImV4cCI6MTYxOTQwNTY5MH0.-gevcGQWoQ5HYA7DyOf6AzOYUN6n2tj91h6S_cIIpCoYT4OxfEZODompX9Njp0QeLuIz-FSCx8w5JGhKjht1Yg', 1000337265)\n\n    @pytest.mark.parametrize(\"case\",cases)\n    def test_api_audit(self,case,get_root_token,get_token):\n        data = self.mid_case.re_repelace(case[\"data\"], 1)\n        excepted = case[\"excepted\"]\n        head = eval(case[\"Headers\"])\n        if head[\"Authorization\"]:\n            head[\"Authorization\"] =  get_token[0]\n        else:\n            head[\"Authorization\"] = get_root_token[1][\"shenhe\"]\n        if case[\"case_id\"] == 3:\n            respose =  self.mid_obj.req_obj.send_request(method=case[\"method\"], url=self.url, data=data, headers=head)\n        else:\n            data = eval(data)\n            respose = self.mid_obj.req_obj.send_request(method=case[\"method\"], url=self.url, json=data, headers=head)\n    \n        #断言\n        try:\n            for k,v in eval(case[\"excepted\"]).items():\n                assert jsonpath.jsonpath(respose, k)[0] == v\n            self.mid_obj.logger.info(\"第{}条测试用例通过,请求参数是{}\".format(case[\"case_id\"], data))\n            self.excel.write_data(row=case[\"case_id\"] + 1, column=9, value=\"pass\")\n        except AssertionError as e:\n            Midd_Ware.logger.info(\"第{}条测试用例不通过 ！\\n 请求体是{} 请求内容是{}。预期结果是{}---返回结果是{}\".format(case[\"case_id\"], data, case[\"title\"],excepted,respose[\"msg\"]))\n            self.excel.write_data(row=case[\"case_id\"] + 1, column=9, value=\"failure\")\n>           raise e\n\ntest_audit.py:45: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <tests.test_audit.Test_Audit object at 0x0000016EF719A128>\ncase = {'Headers': '{\"Content-Type\":\"application/json\",\"X-Lemonban-Media-Type\":\"lemonban.v2\",\"Authorization\":None}', 'auto_script': '项目不存在', 'case_id': 2, 'data': '{\\n    \"loan_id\":\"#N_id#\",\\n    \"approved_or_not\":\"true\"\\n\\n}', ...}\nget_root_token = ({'N_id': 2064567, 'get_id': 2037397, 'give_id': 2052534, 'no_id': 2037790, ...}, {'shenhe': 'Bearer eyJhbGciOiJIUzUxM...5MH0.pIjGHVqRQZEKWvAUBcTKfCPSn2MA2ONc_tmAliMBDgS24mJBnQJIt1F8kN_Qy6vIS2sw_xm1sALG9cyEMzXp-Q', 'shenhe_id': 1000337288})\nget_token = ('Bearer eyJhbGciOiJIUzUxMiJ9.eyJtZW1iZXJfaWQiOjEwMDAzMzcyNjUsImV4cCI6MTYxOTQwNTY5MH0.-gevcGQWoQ5HYA7DyOf6AzOYUN6n2tj91h6S_cIIpCoYT4OxfEZODompX9Njp0QeLuIz-FSCx8w5JGhKjht1Yg', 1000337265)\n\n    @pytest.mark.parametrize(\"case\",cases)\n    def test_api_audit(self,case,get_root_token,get_token):\n        data = self.mid_case.re_repelace(case[\"data\"], 1)\n        excepted = case[\"excepted\"]\n        head = eval(case[\"Headers\"])\n        if head[\"Authorization\"]:\n            head[\"Authorization\"] =  get_token[0]\n        else:\n            head[\"Authorization\"] = get_root_token[1][\"shenhe\"]\n        if case[\"case_id\"] == 3:\n            respose =  self.mid_obj.req_obj.send_request(method=case[\"method\"], url=self.url, data=data, headers=head)\n        else:\n            data = eval(data)\n            respose = self.mid_obj.req_obj.send_request(method=case[\"method\"], url=self.url, json=data, headers=head)\n    \n        #断言\n        try:\n            for k,v in eval(case[\"excepted\"]).items():\n>               assert jsonpath.jsonpath(respose, k)[0] == v\nE               AssertionError: assert '项目不在审核中状态' == '项目不存在'\nE                 - 项目不存在\nE                 + 项目不在审核中状态\n\ntest_audit.py:39: AssertionError"}, "attachments": [{"name": "log", "source": "fef54159-c765-49a9-bc67-67517d5bc653-attachment.txt", "type": "text/plain"}, {"name": "stderr", "source": "3af972d5-2b32-4093-984e-074934be7d4a-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "case", "value": "{'case_id': 2, 'interface': '审核项目', 'url': '/loan/audit', 'method': 'PATCH', 'title': '审核一个不存在的项目ID', 'data': '{\\n    \"loan_id\":\"#N_id#\",\\n    \"approved_or_not\":\"true\"\\n\\n}', 'Headers': '{\"Content-Type\":\"application/json\",\"X-Lemonban-Media-Type\":\"lemonban.v2\",\"Authorization\":None}', 'excepted': '{\"$..code\":2, \"$..msg\": \"项目不存在\"}', 'result': 'pass', 'auto_script': '项目不存在'}"}], "start": 1619405391911, "stop": 1619405392137, "uuid": "7ef0ee33-1561-4e51-ae41-b8e8d80cfabc", "historyId": "db5e13b6cedf58e931ce32243352049e", "testCaseId": "b4fbaf9314ba85c67acc5815b54923c3", "fullName": "test_audit.Test_Audit#test_api_audit", "labels": [{"name": "tag", "value": "@pytest.mark.usefixtures('init_audit')"}, {"name": "suite", "value": "test_audit"}, {"name": "subSuite", "value": "Test_Audit"}, {"name": "host", "value": "DESKTOP-AC1MULO"}, {"name": "thread", "value": "1320-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_audit"}]}